compiler/rustc_metadata/src/rmeta/decoder.rs

```diff
@@ -1099,10 +1099,7 @@ impl<'a, 'tcx> CrateMetadataRef<'a> {
         };
 
         // Iterate over all children.
-        let macros_only = self.dep_kind.lock().macros_only();
```

Comment:
The `macros_only` flag is only set for proc macro crates (processed above), and for `#[no_link]` crates.
Maybe it should be renamed, but I don't want to do it in a single PR with observable language changes.